sources/packages/backend/apps/api/src/route-controllers/application-offering-change-request/application-offering-change-request.institutions.controller.ts

```diff
@@ -68,6 +86,42 @@ export class ApplicationOfferingChangeRequestInstitutionsController extends Base
     };
   }
 
+  /**
+   * Gets an eligible application that can be requested for application
+   * offering change.
+   * @param locationId location id.
+   * @returns eligible application.
+   */
+  @ApiNotFoundResponse({
+    description: "Application not found or it is not eligible.",
+  })
+  @Get("available/application/:applicationId")
+  async getEligibleApplication(
+    @Param("locationId", ParseIntPipe) locationId: number,
+    @Param("applicationId", ParseIntPipe) applicationId: number,
+  ): Promise<ApplicationOfferingChangeSummaryDetailAPIOutDTO> {
+    const application =
+      await this.applicationOfferingChangeRequestService.getEligibleApplications(
+        locationId,
+        { applicationId },
+      );
+    if (!application) {
+      throw new NotFoundException(
+        "Application not found or it is not eligible.",
+      );
+    }
+    return {
+      applicationNumber: application.applicationNumber,
+      applicationId: application.id,
```

Comment:
The applicationId is already supplied to the API input. Should we return as well?

Reply:
It makes sense. I will remove it.

Reply:
Changed.